packages/monaco/src/browser/monaco-command.ts

```diff
@@ -120,6 +124,65 @@ export class MonacoEditorCommandHandlers implements CommandContribution {
             }
         };
     }
+    protected newConfigTabbingHandler(): MonacoEditorCommandHandler {
+        return {
+            execute: (editor: MonacoEditor) => {
+                const run = (type: string) => {
+                    const editorModel = editor.document;
+                    if (editorModel && editorModel.textEditorModel) {
+                        editorModel.textEditorModel.updateOptions({
+                            tabSize: editorModel.textEditorModel.getOptions().tabSize,
+                            insertSpaces: type === 'Spaces'
+                        });
+                    }
+                };
+                const options: QuickOpenItem[] = ['Spaces', 'Tabs'].map(option => new EditorQuickOpenItem(option, `Indent Using ${option}`, run));
+
+                const updateTabSize = () => {
+                    const editorModel = editor.document;
+                    if (editorModel && editorModel.textEditorModel) {
+                        const tabSize = editorModel.textEditorModel.getOptions().tabSize;
+                        const tabSizeOptions = Array.from(Array(8), (_, x) => x + 1)
+                            .map(size => new UpdateTabSizeOpenItem(size === tabSize ? `${size} *` : size.toString(), () => {
+                                editorModel.textEditorModel.updateOptions({
+                                    tabSize: size,
+                                    insertSpaces: editorModel.textEditorModel.getOptions().insertSpaces
+                                });
+                            },
+                                (mode: QuickOpenMode) => mode === QuickOpenMode.OPEN,
+                                () => true)
+                            );
+                        this.open(tabSizeOptions, 'Change the Tab Size to...');
+                    }
+
+                };
+                options.push(new UpdateTabSizeOpenItem('Change Tab Size to...', updateTabSize, (mode: QuickOpenMode) => mode === QuickOpenMode.OPEN, () => false));
+                this.open(options, 'Select Action');
+            },
+            isEnabled: (editor: MonacoEditor) => !!editor
+        };
+    }
+    private open(items: QuickOpenItem | QuickOpenItem[], placeholder: string): void {
+        this.quickOpenService.open(this.getModel(items), this.getOptions(placeholder));
+    }
+    private getModel(items: QuickOpenItem | QuickOpenItem[]): QuickOpenModel {
+        const quickOpenItems = this.getQuickOpenItemsArray(items);
```

Comment:
Also I think `this.getQuickOpenItemsArray(items)` is only called once so maybe this could be replaced by `const quickOpenItems = Array.isArray(items) ? items: [items]` directly.